iothub_client/tests/common_dt_e2e/iothubclient_common_dt_e2e.c

```diff
@@ -905,6 +905,11 @@ void dt_e2e_send_module_id_test(IOTHUB_CLIENT_TRANSPORT_PROVIDER protocol, IOTHU
 
     char *twinData = dt_e2e_get_twin_from_service(serviceClientDeviceTwinHandle, deviceToUse);
 
+    IoTHubDeviceTwin_Destroy(serviceClientDeviceTwinHandle);
```

Comment:
Interesting. Why would valgrind complain about the order? do you have the error messages?

Reply:
no, another issue. without the cleanup, all test cases fail after this